### PR TITLE
doc: add compressdev programmers guide

### PR DESCRIPTION
**Pull request by ashishgupta4**
Add programmer’s guide for compress dev library.

Signed-off-by: Shally Verma <shally.verma@caviumnetworks.com>
Signed-off-by: Sunila Sahu <sunila.sahu@caviumnetworks.com>
Signed-off-by: Ashish Gupta <ashish.gupta@caviumnetworks.com>

### CODE REVIEW

File: doc/guides/prog_guide/compressdev_lib.rst [hunk 1]
```diff
@@ -0,0 +1,984 @@
...
+parameters for socket selection and number of queue pairs.
+
+.. code-block:: c
+
+    struct rte_compressdev_config {
+        int socket_id;
+        /**< Socket to allocate resources on */
+        uint16_t nb_queue_pairs;
+        /**< Number of queue pairs to configure on device */
+    };
```
Comment (pablodelara) [inline doc/guides/prog_guide/compressdev_lib.rst]: Current configuration has more parameters.

File: doc/guides/prog_guide/compressdev_lib.rst [hunk 2]
```diff
@@ -0,0 +1,984 @@
...
+Following are current compression algorithms feature flags:
+
+* Stateful Compression
+* Stateful Decompression
+* Scatter-Gather input/output
+* Adler32 checksum generation
+* CRC32 checksum generation
+* Adler and CRC checksum generation
+* Uncompressed blocks generation
+
```
Comment (pablodelara) [inline doc/guides/prog_guide/compressdev_lib.rst]: Missing hash and shareable priv xform feature flags

File: doc/guides/prog_guide/compressdev_lib.rst [hunk 3]
```diff
@@ -0,0 +1,984 @@
...
+		 */
+		uint8_t status;
+		/**<
+		 * Operation status - use values from enum rte_comp_status.
+		 * This is reset to
+		 * RTE_COMP_OP_STATUS_NOT_PROCESSED on allocation from mempool and
+		 * will be set to RTE_COMP_OP_STATUS_SUCCESS after operation
+		 * is successfully processed by a PMD
+		 */
+	}
```
Comment (pablodelara) [inline doc/guides/prog_guide/compressdev_lib.rst]: Add hash structure

File: doc/guides/prog_guide/compressdev_lib.rst [hunk 4]
```diff
@@ -0,0 +1,984 @@
...
+- RTE_COMP_OP_STATUS_OUT_OF_SPACE_TERMINATED,
+	Output buffer ran out of space during processing. Error case,
+	PMD cannot continue from here.
+
+- RTE_COMP_OP_STATUS_OUT_OF_SPACE_RECOVERABLE,
+	Output buffer ran out of space before operation completed, but this
+	is not an error case. Output data up to op.produced can be used and
+	next op in the stream should continue on from op.consumed+1.
+
+produced,consumed And Operation Status
```
Comment (pablodelara) [inline doc/guides/prog_guide/compressdev_lib.rst]: Capital letter missing, plus whitespace after ",".

File: doc/guides/prog_guide/compressdev_lib.rst [hunk 5]
```diff
@@ -0,0 +1,984 @@
...
+black-listed using the EAL command line options.
+
+Virtual devices can be created by two mechanisms, either using the EAL command
+line options or from within the application using an EAL API directly.
+
+From the command line using the --vdev EAL option
+
+.. code-block:: console
+
+   --vdev  '<pmd name>,max_nb_queue_pairs=2,socket_id=0'
```
[the remark this answers is not in the document]
Comment (1234sv) [reply doc/guides/prog_guide/compressdev_lib.rst]: I was known to max_nb_queue_pair = 0 change for vdevs, however did not notice its removal from pmd_init_parms. What is the purpose of private_data_size in new struct?

File: doc/guides/prog_guide/compressdev_lib.rst (hunk 1 above)
Comment (1234sv) [reply doc/guides/prog_guide/compressdev_lib.rst]: Sorry to miss that update

**Comment by 1234sv** [comment]
All changes have been incorporated. Only open question is regarding priv_data_size in pmd_init_params. Please clarify so that we can update its usage in guide.

**Comment by ashishgupta4** [comment]
Closing this PR . raise new with new doc updated with review inputs.

**Pull request closed without merging**